e2e/README.md

```diff
@@ -145,7 +149,1089 @@ func E2ETests(env e2e.TestEnv) testhelper.Tests {
+
+Make sure to also [mark your test
+appropriately](#parallel-par-vs-non-parallel-seq-tests) if it cannot be run in
+parallel.
+
+**Scenario 2:** You're trying to get the e2e test to only run your test. You've
+set the [E2E_GROUPS and E2E_TESTS](#running-the-e2e-suite) environment
+variables, and you've double checked the spelling of the group & test names, but
+it's not catching your test; nothing is running.
+
+**Common cause:** You're trying to use E2E_TESTS to filter tests by something
+*other* than the top-level test name. Here is an example of a test name from a
+specific e2e subtest:
+
+```plain
+TestE2E/SEQ/INSTANCE/auth/Root/start_before_auth
+```
+
+`TestE2E` is the single top-level name for all e2e tests; next comes [`PAR` or
+`SEQ`](#parallel-par-vs-non-parallel-seq-tests), then comes the group name
+(which is what E2E_GROUPS matches), followed by the top-level test name (which
+is what E2E_TESTS matches; in this case, `auth`). The E2E_TESTS environment
+variable is a regular expression that is matched only against the top-level test
+name.
+
+At the present, there is no way to match against more deeply-embedded components
+of the test path. However, individual top-level tests are usually short enough
+to make this granularity sufficient in practice.
+
+**Another common cause:** You've confused the test function's name for the test
+name, or the test group source file's name for the group name.
+
+The values that E2E_TESTS matches against are the values of the keys in the
+`testhelper.Tests` map. Consider this example, repeated from earlier:
+
+```go
+func E2ETests(env e2e.TestEnv) testhelper.Tests {
+	c := ctx{
+		env: env,
+	}
+
+	np := testhelper.NoParallel
+
+	return testhelper.Tests{
+		"add":            c.remoteAdd,
+		"list":           c.remoteList,
+		"default or not": c.remoteDefaultOrNot,
+		"remove":         c.remoteRemove,
+		"status":         c.remoteStatus,
+		"test help":      c.remoteTestHelp,
+		"use":            c.remoteUse,
+		"use exclusive":  np(c.remoteUseExclusive),
+	}
+}
+```
+
+To run only the c.remoteUseExclusive() test, the environment variable value
+`E2E_TESTS=remoteUseExclusive` won't work. That's because, as far as the testing
+suite is concerned, the test's name is `use exclusive`. Therefore, a correct
+value to run this test only would be, e.g., `E2E_TESTS=use.exclusive`.
+
+In a similar vein, it is important to note that the name of the `.go` file for
+an e2e group is not the same thing as the name of that test group. For example,
+the file e2e/imgbuild/imgbuild.go, which contains the code for the package
+`imgbuild`, is responsible for the e2e group named `BUILD` (and *not*
+`IMGBUILD`!). This can be seen by consulting the `e2eGroups` map defined in
+e2e/suite.go:
+
+```go
+var e2eGroups = map[string]testhelper.Group{
+	"ACTIONS":    actions.E2ETests,
+	"BUILDCFG":   e2ebuildcfg.E2ETests,
+	"BUILD":      imgbuild.E2ETests, // <--- this one!
```

Comment:
There's no real reason not to change this at this point, I don't think. Having to point it out as a gotcha (I think ENV is too) probably shows we should fix it...

Reply:
It's not just these two... it's also "BUILDCFG" vs. "e2ebuildcfg". I wasn't sure there whether there was some reason which I wasn't aware of behind these choices. In any event, that's a significant change in the code; I'd lean towards making that a separate PR, that then also includes removing this part of the discussion here. What do you think?

Reply:
We could also fold the zapping of EnsureORASPrivImage() into that same PR. (The issue there is not just that it's unused code; there's a fundamental problem with the idea behind it, and in any event it's covered by our testing in a different way now.)

Reply:
Yeah, sorry - I was assuming the changes could/would be made in a PR merged ahead of this one. I should have been clear.

If the `EnsureORASPrivImage` is going to be removed, let's have a completely separate PR for that. I'm not clear whether it's a clean backport, and to which versions.... separate PR makes it easier to reason about that.